### DISCUSSION TITLE
What density function does "HAR1"  as prior on coefficients correspond to?

### DISCUSSION THREAD

**Discussion by EricZarahn** [wenjie2wang/dynsurv#4] [Q&A] [answered]
Hello,

First of all, thank you for the dynsurv package.

I am not having any trouble using the code, which is wonderful! However, I would like to know what the prior density on the coefficients is that corresponds to the "HAR1" selection (as exemplified in the code below from the dynsurv documentation). That is, I don't know right now what "HAR1" corresponds to in terms of a known density function.

My current best guess from reading Wang et al. BMC Medical Research Methodology (2016) is that the prior in this case is a Gaussian with mean 0 and variance given by a scaled (by a variable named "ao" in that paper) inverse gamma with the specified shape and scale parameters. Is that right? If so, is "a0" specified/hard-coded somewhere, or does it also have a prior?

  ### Fit dynamic coefficient model with dynamic hazards (in log scales)
  fit3 <- bayesCox(myformula, mydata, model = "Dynamic",
                   base.prior = list(type = "Const"),
                   coef.prior = list(**type = "HAR1", shape = 2, scale = 1**),
                   gibbs = list(iter = num.iter, burn = 20, thin = 1,
                                verbose = F, nReport = 20),
                   control = list(intercept = TRUE))

Many thanks,
Eric

**Comment by EricZarahn** [comment]
I found in the comments of the bayesCox function that a0 defaults to 100:

##'     \item{a0:}{Multiplier for initial variance in time-varying or dynamic
##'         models, default 100;}

So the only thing that remains from my initial question is whether the prior on the coefficients is a Gaussian with mean 0 and variance given a0 times an inverse gamma with the specified shape and scale parameters.

Thanks again,
Eric

**Reply by wenjie2wang** [reply, marked as answer]
Hi Eric,

> So the only thing that remains from my initial question is whether the prior on the coefficients is a Gaussian with mean 0 and variance given a0 times an inverse gamma with the specified shape and scale parameters.

The prior specifications for the dynamic model are given as follows:

![Screenshot from 2022-08-20 21-58-35](https://user-images.githubusercontent.com/11081955/185772048-f256add4-5581-47e7-af32-cda77e83e328.png)

where $\{\tau_j\}$'s are jump points, $J$ is the number of jumps, and $a_0$ is a hyperparameter.  See Section 3 of Wang et al., (2013) for details.

- Wang, X., Chen, M., & Yan, J. (2013). Bayesian dynamic regression models for interval censored survival data with application to children dental health. Lifetime Data Analysis, 19(3), 297–316.

**Reply by EricZarahn** [reply]
Thank you for the answer Wenjie.

I have another question if you would be so kind: I am doing simulations using in which I generate data with right-censoring for 2 groups of subjects with hazard rate of 0.03 in group A (reference level of factor) vs. 0.01 group B and then estimate the hazard function using dynsurv::bayesCox. 

I am finding that, for a fixed sample size of group A, that the estimated hazard ratio decreases from the simulated value of 3 as I increase the sample size of group B. This phenomenon is surprising to me, as I was expecting the estimated hazard ratio to, if anything, become closer to ground truth as I increased the sample size of either group.

Here are some simulation results for N = 50 realizations of data per condition:

![image](https://user-images.githubusercontent.com/36579858/187087802-123a15a1-291e-421e-b1c9-001c34eebdb1.png)


This is the call I used:

      fit <- bayesCox(myformula, mydata, model = "Dynamic",
                       base.prior = list(type = "Const"),
                       coef.prior = list(type = "HAR1", shape = 2, scale = 1),
                       gibbs = list(iter = 1000, burn = 20, thin = 1,
                                    verbose = F, nReport = 20),
                       control = list(intercept = TRUE))    

Can you provide any insight?

Many thanks again,
Eric

**Reply by wenjie2wang** [reply]
The dynamic model is intended for interval-censored data, some of which can be right-censored [t, $\infty$).  In general, it would not be a good idea to apply it to right-censored data with exact event times as all the unique event times will be candidates for the potential jumps.

**Reply by EricZarahn** [reply]
Ok, thank you very much for both answers Wenjie!

Best,
Eric
